src/main/java/seedu/address/model/ItemModelManager.java

```diff
@@ -234,4 +253,68 @@ public void clear() {
     public void sort() {
         this.visualList = visualList.sort();
     }
+
+    /**
+     * Enable and disable the priority mode
+     * @return a boolean value. If true, means priority mode is on, else returns false.
+     * @throws IllegalListException if the visualList is not a task list.
+     */
+    public boolean togglePriorityMode() throws IllegalListException {
+        if (!(visualList instanceof TaskList)) {
+            throw new IllegalListException();
+        }
+
+        priorityMode = !priorityMode;
+        if (!priorityMode) {
+            sortedTask = null;
+            this.visualList = taskList;
+        } else {
+            sortedTask = new PriorityQueue<>((item1, item2) -> {
+                Task task1 = item1.getTask().get();
+                Task task2 = item2.getTask().get();
+                return task1.getPriority().compareTo(task2.getPriority());
```

Comment:
Great work! But just wondering, I thought getPriority() returns HIGH/MEDIUM/LOW? How is that compared?

Reply:
It's compared with the ordinal value of the enum!
https://docs.oracle.com/javase/7/docs/api/java/lang/Enum.html